barpyrus/trayer.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+import threading
+
+from barpyrus.widgets import Widget
+from barpyrus.core import EventInput
+from Xlib.display import Display, X
+
+class TrayerWatch(EventInput):
+    def __init__(self, args = { }):
+        command = [ 'trayer' ]
+        self.default_args = {
+            'edge': 'top',
+            'align': 'right',
+            'widthtype': 'request',
+            'expand': 'true',
+            'SetDockType': 'true',
+            'transparent': 'true',
+            'alpha': '0',
+            'height': '16',
+            'margin': '0',
+            'tint': '0x29b2e',
+        }
+        for key,val in args.items():
+            self.default_args[key] = val
+        for key,val in self.default_args.items():
+            command += ["--%s" % (key), str(val)]
+
+        super(TrayerWatch,self).__init__(command)
+        self.proc.stdin.close()
+        self.proc.stdout.close()
```

Comment:
Let us hope trayer does not get killed by sigpipe :-)

Reply:
Mhh, good point. I might have gone over the top there. However, I did not notice any issues so far (maybe some longterm tests will prove me wrong :) )

Reply:
Maybe they handle sigpipe or maybe they don't write to stdout (they probably don't have to)